flink-table/flink-table-runtime-blink/src/main/java/org/apache/flink/table/filesystem/stream/PartitionCommitPredicate.java

```diff
@@ -0,0 +1,64 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.table.filesystem.stream;
+
+import org.apache.flink.configuration.Configuration;
+import org.apache.flink.streaming.runtime.tasks.ProcessingTimeService;
+
+import java.util.List;
+
+import static org.apache.flink.table.filesystem.FileSystemOptions.SINK_PARTITION_COMMIT_TRIGGER;
+
+/**
+ * Partition commit predicate. See {@link PartitionTimeCommitPredicate}. See {@link
+ * ProcTimeCommitPredicate}
+ */
+public interface PartitionCommitPredicate {
+    String PARTITION_TIME = "partition-time";
+    String PROCESS_TIME = "process-time";
+
+    boolean isPartitionCommittable(String partition, long creationTime, long watermark);
```

Comment:
Maybe provide a context?
```
PredicateContext {
    String partition();
    long createProcTime();
    long currentProcTime();
    long currentWatermark();
}
```
This will make it clearer and avoid me guessing which time is `creationTime`.
- partition corresponding to createProcTime.
- currentProcTime corresponding to currentWatermark.
- And you can throw exception in `PartitionTimeCommitTrigger` for `createProcTime`, throw exception in `ProcTimeCommitTrigger` for `currentWatermark`, this is better than default zero.

Reply:
Thanks @JingsongLi 's advice.
Now, I add the code to provide the PredicateContext.
Look forward your review again.